File: Server/DataBase/Entities/RecipeEnergyEntity.cs
namespace Server.DataBase.Entities;

public class RecipeEnergyEntity
{
    public int Id { get; init; }
    public int RecipeEntityId { get; init; }
    public int CaloriesFrom { get; set; }
    public int CaloriesTo { get; set; }
    public int FatsFrom { get; set; }
    public int FatsTo { get; set; }
    public int CarbohydratesFrom { get; set; }
    public int CarbohydratesTo { get; set; }
    public int ProteinsFrom { get; set; }
    public int ProteinsTo { get; set; }
    public RecipeEntity RecipeEntity { get; set; }
}

File: client/src/models/RecipeEnergyEntity.js
class RecipeEnergyEntity {
    constructor(json) {
        if (json) {
            this.caloriesFrom = json.caloriesFrom;
            this.caloriesTo = json.caloriesTo;
            this.fatsFrom = json.fatsFrom;
            this.fatsTo = json.fatsTo;
            this.carbohydratesFrom = json.carbohydratesFrom;
            this.carbohydratesTo = json.carbohydratesTo;
            this.proteinsFrom = json.proteinsFrom;
            this.proteinsTo = json.proteinsTo;
        } else {
            this.caloriesFrom = 0;
            this.caloriesTo = 0;
            this.fatsFrom = 0;
            this.fatsTo = 0;
            this.carbohydratesFrom = 0;
            this.carbohydratesTo = 0;
            this.proteinsFrom = 0;
            this.proteinsTo = 0;
        }
    }
}

export default RecipeEnergyEntity;